plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -1004,13 +1004,20 @@ Map<String, String> getColumnHistograms(JdbcTableHandle table)
             }
 
             RemoteTableName remoteTableName = table.getRequiredNamedRelation().getRemoteTableName();
-            return handle.createQuery("" +
-                            "SELECT COLUMN_NAME, HISTOGRAM FROM INFORMATION_SCHEMA.COLUMN_STATISTICS " +
-                            "WHERE SCHEMA_NAME = :schema AND TABLE_NAME = :table_name")
-                    .bind("schema", remoteTableName.getCatalogName().orElse(null))
-                    .bind("table_name", remoteTableName.getTableName())
-                    .map((rs, ctx) -> new SimpleEntry<>(rs.getString("COLUMN_NAME"), rs.getString("HISTOGRAM")))
-                    .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+            try {
+                return handle.createQuery("" +
+                                "SELECT COLUMN_NAME, HISTOGRAM FROM INFORMATION_SCHEMA.COLUMN_STATISTICS " +
+                                "WHERE SCHEMA_NAME = :schema AND TABLE_NAME = :table_name")
+                        .bind("schema", remoteTableName.getCatalogName().orElse(null))
+                        .bind("table_name", remoteTableName.getTableName())
+                        .map((rs, ctx) -> new SimpleEntry<>(rs.getString("COLUMN_NAME"), rs.getString("HISTOGRAM")))
+                        .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+            }
+            catch (Exception e) {
+                log.debug("INFORMATION_SCHEMA.COLUMN_STATISTICS table is not available: %s", e);
+                // When some service provider databases modified based on MySQL do not keep these tables, the corresponding data cannot be obtained, such as AnalyticDB for MySQL.
```

Comment:
We don't support AnalyticDB. This indicates a new dedicated connector is needed like MariaDB & SingleStore connectors.

Reply:
Some secondary development databases based on the MySQL protocol cannot be supported, because some secondary developers have deleted some tables used by meta, but they can be queried based on mysql-jdbc. If restrictions are made here, these databases cannot be supported.